### PR TITLE
Use Sepolia chain ID

### LINKED ISSUE #389: Instances of SN_GOERLI chain ID need to be replaced with SN_SEPOLIA
Currently we depend on starknet-rs' implementation of ChainId, which doesn't have this updated yet. We could incite the update of starknet-rs by contacting its maintainer, or we could implement our own ChainId abstraction.

Also, in the Display implementation, the needed change is independent of starknet-rs.

Labeling as enhancement and as bug since Goerli is no longer supported.

### LINKED ISSUE #400: Integration tests refactor
Remove assertion on transaction_hash in integration tests, because with each new blockifier version the cost of a transaction is changed. This affects the transaction hash. Instead check if the transaction is successful

### PR DESCRIPTION
**Pull request by FabijanC**
## Usage related changes

- Close #389

## Development related changes

- Close #400

## Checklist:

- [x] Applied formatting - `./scripts/format.sh`
- [x] No linter errors - `./scripts/clippy_check.sh`
- [x] No unused dependencies - `./scripts/check_unused_deps.sh`
- [x] Performed code self-review
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [x] Documented the changes
- [x] Linked the issues which this PR resolves
- [x] Checked the TODO section in README.md if this PR resolves it
- [x] Updated the tests
- [x] All tests are passing - `cargo test`

### CODE REVIEW

File: crates/starknet-devnet-core/src/starknet/add_l1_handler_transaction.rs [hunk 1]
```diff
@@ -95,7 +95,7 @@ mod tests {
         .with_hash(ChainId::Testnet.to_felt());
 
         let transaction_hash = Felt::from_prefixed_hex_str(
-            "0x6182c63599a9638272f1ce5b5cadabece9c81c2d2b8f88ab7a294472b8fce8b",
+            "0x1b24ea8dd9e0cb603043958b27a8569635ea13568883cc155130591b7ffe37a",
         )
```
Comment (FabijanC) [inline crates/starknet-devnet-core/src/starknet/add_l1_handler_transaction.rs:99]: This could be the only remaining place where we have a hardcoded tx hash, but I didn't know what to do with the test.

File: crates/starknet-devnet/tests/get_transaction_by_hash.rs [hunk 2]
```diff
@@ -42,7 +42,7 @@ mod get_transaction_by_hash_integration_tests {
             &devnet.json_rpc_client,
             signer,
             account_address,
-            chain_id::TESTNET,
+            chain_id::SEPOLIA,
```
Comment (marioiordanov) [inline crates/starknet-devnet/tests/get_transaction_by_hash.rs:45]: although its done by another PR, why not change it to use constants::CHAIN_ID

Comment (FabijanC) [reply crates/starknet-devnet/tests/get_transaction_by_hash.rs:45]: what other PR?

File: crates/starknet-devnet-core/src/starknet/add_l1_handler_transaction.rs (hunk 1 above)
Comment (marioiordanov) [reply crates/starknet-devnet-core/src/starknet/add_l1_handler_transaction.rs:99]: its ok, because each of the parameters is hardcoded

File: crates/starknet-devnet/tests/get_transaction_by_hash.rs (hunk 2 above)
Comment (FabijanC) [reply crates/starknet-devnet/tests/get_transaction_by_hash.rs]: Addressed in https://github.com/0xSpaceShard/starknet-devnet-rs/pull/396/commits/afc3ae3ac3d0faf89a41f402528504934dc5833e

**Review by marioiordanov — APPROVED** [review]
